frontend/src/components/entities/create-field-dialog.tsx

```diff
@@ -0,0 +1,441 @@
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from "@/components/ui/form"
+import { Input } from "@/components/ui/input"
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "@/components/ui/select"
+import { Textarea } from "@/components/ui/textarea"
+import {
+  Tooltip,
+  TooltipContent,
+  TooltipProvider,
+  TooltipTrigger,
+} from "@/components/ui/tooltip"
+
+const fieldTypes: {
+  value: FieldType
+  label: string
+  icon: React.ElementType
+}[] = [
+  { value: "TEXT", label: "Text", icon: Type },
+  { value: "INTEGER", label: "Integer", icon: Hash },
+  { value: "NUMBER", label: "Number", icon: CircleDot },
+  { value: "BOOL", label: "Boolean", icon: ToggleLeft },
+  { value: "JSON", label: "JSON", icon: Braces },
+  { value: "DATE", label: "Date", icon: Calendar },
+  { value: "DATETIME", label: "Date and time", icon: CalendarClock },
+  { value: "SELECT", label: "Select", icon: SquareCheck },
+  { value: "MULTI_SELECT", label: "Multi-select", icon: ListTodo },
+]
+
+const createFieldSchema = z.object({
+  key: z
+    .string()
+    .min(1, "Field key is required")
+    .regex(
+      /^[a-z][a-z0-9_]*$/,
+      "Field key must start with a letter, be lowercase, and contain only letters, numbers, and underscores"
+    ),
+  type: z.enum([
+    "TEXT",
+    "INTEGER",
+    "NUMBER",
+    "BOOL",
+    "JSON",
+    "DATE",
+    "DATETIME",
+    "SELECT",
+    "MULTI_SELECT",
+  ] as const),
+  display_name: z.string().min(1, "Display name is required"),
+  description: z.string().optional(),
+  default_value: z.any().optional(),
+  options: z.array(z.string()).optional(),
+})
+
+type CreateFieldFormData = z.infer<typeof createFieldSchema>
+
+// Primitive field types that support default values
+const PRIMITIVE_FIELD_TYPES: FieldType[] = [
+  "TEXT",
+  "INTEGER",
+  "NUMBER",
+  "BOOL",
+  "JSON",
+  "SELECT",
+  "MULTI_SELECT",
+]
+
+interface CreateFieldDialogProps {
+  open: boolean
+  onOpenChange: (open: boolean) => void
+  onSubmit: (data: EntityFieldCreate) => Promise<void>
+  errorMessage?: string
+}
+
+export function CreateFieldDialog({
+  open,
+  onOpenChange,
+  onSubmit,
+  errorMessage,
+}: CreateFieldDialogProps) {
+  const [isSubmitting, setIsSubmitting] = useState(false)
+  const [submitError, setSubmitError] = useState<string | null>(null)
+
+  const form = useForm<CreateFieldFormData>({
+    resolver: zodResolver(createFieldSchema),
+    defaultValues: {
+      key: "",
+      type: "TEXT" as const,
+      display_name: "",
+      description: "",
+      default_value: "",
+      options: [],
+    },
+  })
+
+  const fieldType = form.watch("type")
+  const supportsPrimitive = useMemo(
+    () => PRIMITIVE_FIELD_TYPES.includes(fieldType),
+    [fieldType]
+  )
+  const isSelectField = useMemo(
+    () => fieldType === "SELECT" || fieldType === "MULTI_SELECT",
+    [fieldType]
+  )
+
+  const handleSubmit = async (data: CreateFieldFormData) => {
+    setIsSubmitting(true)
```

Comment:
prefer to use react-hook-form builtin form.errors or react query APIs to handle form submission states